### ISSUE TITLE
Read-only user can see the sighting source entry

### ISSUE THREAD

**Issue by djha983** [MISP/MISP#5712]
**Is your feature request related to a problem? Please describe.**
I have made few users ReadOnly and they are able to see the list and search attributes and details.All that is fine.I have one issue in sightings.Allowed users add sightings for indicators along with the sighting source which is useful for users within the origination.Sighting source might contain some detail of the device,system where indicator is seen.I create ReadOnly user to provide access to guest user some time and they are able to see those sighting source details also which i don't want them to see.

**Describe the solution you'd like**
Is it possible to create user profile such a way that they get read only access and are restricted to not to see sighting source details.

**Describe alternatives you've considered**
I don't see any alternative yet.

**Additional context**
this will help us to avoid data leak.

**Comment by mokaddem** [comment]
Hello,
Please have a look at `/servers/serverSettings/Plugin` then the `Sightings` tab.
You might find options interesting for you.
Cheers!

**Comment by djha983** [comment]
Hi,
Thank you .
I tried that BUT did not help. May be I am missing something.

My problem is when I login as read-only user, and I go to sighting toplists tab under Global Settings/statistics option, There I can see the source listed.
My requirement is to hide statistics option for read-only or guest user from the Global Actions menu.Is it possible? If not is there any setting to hide sightings toplists tab from statistics page?


[cid:image002.png@01D60217.3F9D5FF0]


[cid:image001.png@01D60215.443B15E0]

Regards,
Deepak Jha

**Comment by mokaddem** [comment]
Indeed, if a user can have access to the sighting data (he belongs to the same organisation that created the sighting - which is the default behavior), he'll be able to see the sources.

**Comment by djha983** [comment]
Hello Sami,

OK.In that case is it possible that I restrict read only user to not to see sightings detail. Or disallow read-only user to user to see the statistics page ?
My requirement is to hide statistics page for some of the read-only users. How can I achieve?

[cid:image001.png@01D6039A.B185C830]

Regards,
Deepak Jha

**Comment by djha983** [comment]
Awaiting response,My requirement is i want to restrict users from seeing statistics.

**Comment by iglocska** [comment]
This is currently a distinction we do not make in the existing role model. Read only users cannot create data, but the data access is rules by the organisation and distribution rules. We can keep the issue open for anyone wanting to use it as inspiration on further refining the access controls in this way for a pull request, but unless it gains some serious traction from the community it is not something that we'll have on our immediate roadmap as it goes against the current ACL strategy.

Thank you for your understanding.

**Comment by djha983** [comment]
Hello Andras,

Thank You for response.
Is there any other role or other way where I can restrict users to not to see the statistics.

Regards,
Deepak Jha
________________________________

**Comment by iglocska** [comment]
No, the statistics is visible to all roles.
